installer-tests/run_tests.py

```diff
@@ -0,0 +1,155 @@
+import glob
+import os
+import pexpect
+import subprocess
+import re
+
+def _expect(line, current_cmd, do_re_escape=True, do_detect_slow=True, strip_comments=True,
+            verbose=True):
+    timeout = 1
+    if do_detect_slow:
+        if line.startswith('$[slow]'):
+            print 'Slow line...'
+            timeout = 30
+            line = line.replace('$[slow]', '', 1)
+
+    if verbose:
+        print 'expecting', line
+
+    if do_re_escape:
+        line = re.escape(line)
+
+    current_cmd.expect(line, timeout=timeout)
+
+
+TEST_ROOT=os.path.dirname(os.path.abspath(__file__))
+SANDSTORM_SOURCE_TREE=os.path.abspath(os.path.join(os.path.abspath(__file__), '..', '..'))
+SANDSTORM_IN_HOME=os.path.join(os.environ['HOME'], 'sandstorm')
+
+def vagrant_destroy():
+    subprocess.check_output(['vagrant', 'destroy', '-f'], cwd=TEST_ROOT)
+
+def handle_test_script(lines):
+    current_cmd = None
+    next_timeout = 1
+
+    for line in lines:
+        # Figure out what we want to do, given this line.
+        if line.startswith('$[run]'):
+            arg = line.replace('$[run]', '')
+            arg = 'vagrant ssh -c "' + arg + '"'
+            print 'starting', arg, '...'
+            current_cmd = pexpect.spawn(arg, cwd=TEST_ROOT)
+        elif '$[exitcode]' in line:
+            left, right = map(lambda s: s.strip(), line.split('$[exitcode]'))
+            # Expect end of file.
+            current_cmd.expect(pexpect.EOF, timeout=1)
+            current_cmd.close()
+            assert current_cmd.exitstatus == int(right)
+
+        elif '$[type]' in line:
+            # First, we expect the left side.
+            left, right = map(lambda s: s.strip(), line.split('$[type]'))
+            _expect(left, current_cmd=current_cmd)
+
+            # Then we sendline the right side.
+            current_cmd.sendline(right)
+        else:
+            # For now, assume the action is expect.
+            _expect(line, current_cmd=current_cmd)
+
+
+def handle_headers(headers_list):
+    postconditions = []
+    cleanups = []
+
+    for header in headers_list:
+        key, value = map(lambda s: s.strip(), header.split(':'))
+        key = key.lower()
+        if key == 'title':
+            print 'Test title:', value
+
+        if key == 'vagrant-destroy-if-bash':
+            print 'hmm'
+            full_bash_cmd = 'if [ %s ] ; then exit 0 ; else exit 1 ; fi' % (value,)
+            exitcode = subprocess.call(['vagrant', 'ssh', '-c', full_bash_cmd], cwd=TEST_ROOT)
+            if exitcode == 0:
+                print 'Destroying...'
+                vagrant_destroy()
+                print 'Recreating...'
+                vagrant_up()
+
+        if key == 'vagrant-precondition-bash':
+            full_bash_cmd = 'if [ %s ] ; then exit 0 ; else exit 1 ; fi' % (value,)
```

Comment:
Is it `exit [ %s ]` or `exit %s` ?

To avoid having to think about that, I wrote it the above way.

FWIW:

```
asheesh@laptop:~ $ if [ -f /tmp ] ; then echo yes ; else echo no ; fi
no
asheesh@laptop:~ $ exit [ -f /tmp ]
exit
bash: exit: [: numeric argument required
```

So I don't think literally `exit [ %s ]` works, but if I'm mistaken by all means let me know.